### PR TITLE
Combine survey and flexible-survey tasks

### PR DESCRIPTION
**Pull request by amyrebecca**
These were only implemented separately in order to reduce risk of regressive impact in the short term. We're now ready to combine the two, as well as fix some outstanding bugs.

### CODE REVIEW

**Comment by amyrebecca** [comment]
Ali decided that if we couldn't get all of the columns to be equal length, it was better to just stick with horizontal wrapping, so the last commit removes all of the style changes for drawing the choices in columns

**Comment by brian-c** [comment]
Is the `columnize` function from Serengeti not working in this case? I thought we were gonna try that.

https://github.com/zooniverse/Serengeti/blob/master/app/lib/columnize.coffee

**Comment by amyrebecca** [comment]
No, the only guidance I got on this was that it should be do-able with flex box, by using the data-breakpoint stuff you had set up

**Comment by aliburchard** [comment]
Pretty sure we talked about writing custom functions to handle the dynamic lists: https://github.com/zooniverse/Panoptes-Front-End/issues/2085

**Comment by amyrebecca** [comment]
weird, i see the discussion but i never got alerts about those messages.

unfortunately, i've got to move on to the geordi integration tasks, so the columns are not going to make it, certainly not for a launch on thursday :/

**Comment by srallen** [comment]
@brian-c, do you have time today or tomorrow to review and merge? Wisconsin Wildlife Watch needs to get its second season launched this week.

File: app/pages/lab/workflow.cjsx [hunk 1]
```diff
@@ -55,7 +55,8 @@ EditWorkflowPage = React.createClass
     @history.createHref("/projects/#{owner}/#{name}/classify", viewQuery)
 
   canUseTask: (project, task)->
-    task in project.experimental_tools
+    # task in project.experimental_tools
+    true
```
Comment (brian-c) [inline app/pages/lab/workflow.cjsx]: Is this left over from dev?

**Comment by brian-c** [comment]
Checking this out now.

@amyrebecca or @aliburchard, what's the main difference with the flexible task? Looks like most of the changes are in the CSV import functions, has the format of the CSVs changed?

**Comment by brian-c** [comment]
I think I've got it all sorted now. Just a couple minor things:

- Some sub-lists in the CSVs are comma separated (choices' images, characteristics' icons) and some are semicolon separated (questions' answers, includes, excludes). We should pick one.

- `questionsMap` seems to map a question to the choices that need that question. Could that information not be stored on the question itself? Minor, but it'll make updates easier.

@aliburchard I'll get the columns sorted in a separate PR after this one is merged.

**Comment by amyrebecca** [comment]
Thanks for the review! To reply to your concerns:

1. The commented out line and the "return true" was indeed from debugging; I reverted it
2. Good point; I've made this change
3. questionsMap is separate from the individual question because it not only lists which questions map to a particular choice, but also which order they should be presented in, so I think this is probably okay but I will change it if you feel strongly.

**Comment by brian-c** [comment]
Looks good, thanks!

Sounds like #3 is fine if we want to customize question order.

Just had one more thought: are there any projects still using `flexibleSurvey`? We could prevent them from breaking by setting `flexibleSurvey: require './survey'`, but ideally we've converted them already.

**Comment by amyrebecca** [comment]
We are going to be stuck launching WWW Season 2 as flexible survey for various reasons so I think including that line will probably be wise; should that line you suggest be in tasks/index.coffee ?

**Comment by brian-c** [comment]
Yep, tasks/index. Since these are combined now, shouldn't updating WWW just involve change the task `type`, which we can do manually? Or are there considerations beyond the front end (e.g. aggregation) we need to take into account?

**Comment by amyrebecca** [comment]
There aren't other considerations afaik, I just haven't tampered with the production database yet so this seems like the more natural fix :)

**Comment by aliburchard** [comment]
Hey @brian-c and @amyrebecca what's the status of this? Just now getting caught up, but looking forward to having the tasks merged.

**Comment by brian-c** [comment]
Will merge this morning, and then I'm gonna have a stab at implementing Serengeti's sorting function.

**Pull request merged**